makani/models/noise.py

```diff
@@ -99,23 +102,24 @@ def reset(self, batch_size=None):
     # this routine generates a noise sample for a single time step and updates the state accordingly, by appending the last time step
     def update(self, replace_state=False, batch_size=None):
 
-        # create single occurence
-        with torch.no_grad():
-            if batch_size is None:
-                batch_size = self.state.shape[0]
-            newstate = torch.empty((batch_size, self.num_time_steps, self.num_channels, self.lmax_local, self.mmax_local, 2), dtype=self.state.dtype, device=self.state.device)
-            if self.state.is_cuda:
-                newstate.normal_(mean=0.0, std=1.0, generator=self.rng_gpu)
-            else:
-                newstate.normal_(mean=0.0, std=1.0, generator=self.rng_cpu)
+         if replace_state:
```

Comment:
so the replace_state was not really used here, is that right? it seems correct to me now. did we never use this?

Reply:
Ok, I see the issue. replace_state was introduced for stateful noise. So this is only relevant for diffusion noise sources. In this case, replace_state = True means that you want to overwrite what is in state. Now, for all other sources, update should always update the state. Can you undo this change? Or do you run into trouble when you do?

Reply:
Reverted.